validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java

```diff
@@ -0,0 +1,114 @@
+
+  private static final String LOGGROUPPATH = "/aws/ecs/otlp/%s/logs";
+
+  protected CloudWatchService cloudWatchService;
+  private static final int CHECK_INTERVAL_IN_MILLI = 30 * 1000;
+  private static final int CHECK_DURATION_IN_SECONDS = 2 * 60;
+  private static final int MAX_RETRY_COUNT = 12;
+  private static final int QUERY_LIMIT = 100;
+  private JsonSchema schema;
+  protected String logGroupName;
+
+  private Context context;
+
+  protected final ObjectMapper mapper = new ObjectMapper();
+
+  @Override
+  public void init(
+      Context context,
+      ValidationConfig validationConfig,
+      ICaller caller,
+      FileConfig expectedDataTemplate)
+      throws Exception {
+    this.context = context;
+    cloudWatchService = new CloudWatchService(context.getRegion());
+    logGroupName = String.format(LOGGROUPPATH, context.getTestingId());
+    MustacheHelper mustacheHelper = new MustacheHelper();
+    String templateInput = mustacheHelper.render(expectedDataTemplate, context);
+    JsonNode jsonNode = JsonLoader.fromString(templateInput);
+    JsonSchemaFactory jsonSchemaFactory =
+        JsonSchemaFactory.newBuilder()
+            .setReportProvider(new ListReportProvider(LogLevel.INFO, LogLevel.FATAL))
+            .freeze();
+    JsonSchema jsonSchema = jsonSchemaFactory.getJsonSchema(jsonNode);
+    this.schema = jsonSchema;
+    caller.callSampleApp();
+  }
+
+  @Override
+  public void validate() throws Exception {
+    RetryHelper.retry(
+        getMaxRetryCount(),
+        CHECK_INTERVAL_IN_MILLI,
+        true,
+        () -> {
+          Instant startTime =
+              Instant.now().minusSeconds(CHECK_DURATION_IN_SECONDS).truncatedTo(ChronoUnit.MINUTES);
+          fetchAndValidateLogs(startTime);
+        });
+  }
+
+  protected void fetchAndValidateLogs(Instant startTime) throws Exception {
+    List<OutputLogEvent> logEvents =
+        cloudWatchService.getLogs(
+            logGroupName, logStreamName, startTime.toEpochMilli(), QUERY_LIMIT);
+    if (logEvents.isEmpty()) {
+      throw new BaseException(
+          ExceptionCode.LOG_FORMAT_NOT_MATCHED,
+          String.format(
+              "[StructuredLogValidator] no logs found under log stream %s" + " in log group %s",
+              logStreamName, logGroupName));
+    }
+    for (OutputLogEvent logEvent : logEvents) {
+      if (logEvent.getMessage().contains("Executing outgoing-http-call")) {
+        validateJsonSchema(logEvent.getMessage());
+      }
+    }
+  }
+
+  protected void validateJsonSchema(String logEventMsg) throws Exception {
+    JsonNode logEventNode = mapper.readTree(logEventMsg);
+    if (schema != null) {
+      ProcessingReport report = schema.validate(JsonLoader.fromString(logEventNode.toString()));
+      if (report.isSuccess()) {
+        log.info("Report was a success");
+      } else {
+        log.info("[StructuredLogValidator] failed to validate schema \n");
+        log.info(report.toString() + "\n");
+      }
```

Comment:
How will the test case know to fail if it reaches this logical block? Nothing is returned back to the calling function to indicate a failure.

It would be much better if you added some unit tests for this new validator class where we could mock out some scenarios.